File: app/src/main/java/org/worshipsongs/service/MobileNetworkService.java
package org.worshipsongs.service;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

/**
 * author:Madasamy
 * version:1.2.0
 */
public class MobileNetworkService implements IMobileNetworkService
{
    @Override
    public boolean isWifi(Object object)
    {
        //object should be system service
        ConnectivityManager connectivityManager = (ConnectivityManager) object;
        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
        if (networkInfo != null) {
            if (networkInfo.isConnected()) {
                if ((networkInfo.getType() == ConnectivityManager.TYPE_WIFI)) {
                    return true;
                }
            }
        }
        return false;
    }

    @Override
    public boolean isMobileData(Object object)
    {
        ConnectivityManager connectivityManager = (ConnectivityManager) object;
        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
        if (networkInfo != null) {
            if (networkInfo.isConnected()) {
                if ((networkInfo.getType() == ConnectivityManager.TYPE_MOBILE)) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: app/src/main/java/org/worshipsongs/service/MobileNetworkService.kt
package org.worshipsongs.service

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

/**
 * author:Madasamy
 * version:1.2.0
 */
class MobileNetworkService : IMobileNetworkService
{
    override fun isWifi(`object`: Any): Boolean
    {
        //object should be system service
        val connectivityManager = `object` as ConnectivityManager
        val networkInfo = connectivityManager.activeNetworkInfo
        return networkInfo != null && networkInfo.isConnected &&
                networkInfo.type == ConnectivityManager.TYPE_WIFI
    }

    override fun isMobileData(`object`: Any): Boolean
    {
        val connectivityManager = `object` as ConnectivityManager
        val networkInfo = connectivityManager.activeNetworkInfo
        return networkInfo != null && networkInfo.isConnected &&
                networkInfo.type == ConnectivityManager.TYPE_MOBILE

    }
}
